shell/server/identity.js

```diff
@@ -0,0 +1,38 @@
+// Sandstorm - Personal Cloud Sandbox
+// Copyright (c) 2016 Sandstorm Development Group, Inc. and contributors
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const Capnp = Npm.require("capnp");
+const IdentityRpc = Capnp.importSystem("sandstorm/identity-impl.capnp");
+
+class IdentityImpl {
+  constructor(identityId, persistentMethods) {
+    _.extend(this, persistentMethods);
+    this.identityId = identityId;
+  }
+};
+
+makeIdentity = (identityId, persistentMethods, requriements) => {
```

Comment:
I guess so.  I comment about this mostly because the more globals we have the harder it is to unwind their messy interdependencies, and because I think we should be moving away from having globals everywhere.  In this case, I doubt there will be much additional use of the code elsewhere (especially in packages), so that weighs in your favor.

One concern I have with your suggestion is that it can be challenging to produce such patches in the face of general discouragement of large, sweeping changes that can be difficult to review and which may produce significant merge conflicts, particularly if not immediately addressed.  (Larger changes tend to take longer to get merged as well.)  If I later write the proposed all-at-once fix, can I expect it to get merged in a timely fashion?

I have been looking about general shell refactoring a fair bit in my personal time, and as a more general rule, it is infeasible to "just" fix all of our globals at once, due to the fact that right now we rely on having an unsafe load order between circular dependencies.  If a shell-level global is called by code in a package (which is strictly unsafe and an inverted dependency, but happens to work at runtime due to lenient JS scoping rules), we will need to move the whole package out to the shell proper before we will be able to remove the global.  But other packages may depend on that package, so we have to first migrate packages loaded last to the shell, before we can move the dependency packages to the shell.  This means that sandstorm-db will likely be one of the last things to move out of a package into `/imports`.

Anyway.  Yeah, I guess you can leave this as-is for now.